independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/api/WebClientApplicationException.java

```diff
@@ -22,18 +22,41 @@
 /**
  * Subclass of {@link WebApplicationException} for use by clients, which forbids setting a response that
  * would be used by the server.
- * FIXME: I'd rather this be disjoint from WebApplicationException, so we could store the response info
- * for client usage. Perhaps we can store it in an alternate field?
+ * FIXME: I'd rather this be disjoint from WebApplicationException, but changing the exception type could break existing
+ * clients catching this exception.
  */
 @SuppressWarnings("serial")
 public class WebClientApplicationException extends WebApplicationException implements ResteasyReactiveClientProblem {
 
+    /**
+     * The response received by the client after making the request.
+     * Saved in an alternate field to prevent it from being used by the server.
+     */
+    private final transient Response clientResponse;
+
     public WebClientApplicationException(int responseStatus) {
-        this(responseStatus, null);
+        this(responseStatus, (String) null);
     }
 
     public WebClientApplicationException(int responseStatus, String responseReasonPhrase) {
         super("Server response is: " + responseStatus, null, new DummyResponse(responseStatus, responseReasonPhrase));
+        this.clientResponse = getResponse();
+    }
+
+    public WebClientApplicationException(Response clientResponse) {
+        super("Server response is: " + clientResponse.getStatus(), new DummyResponse(clientResponse.getStatus(),
```

Comment:
Preserved the dummy response in this case, so that it's not null and breaks existing code that expects to read the status code from the `response` field.

Reply:
This is not right, it will set a `Response` property from the client's response, which is a security risk. The `super.response` field must always be `null`, and the response should be stored in the `clientResponse` property only.

Reply:
:thinking: That's how it was done before my change, and it looks like that's why the DummyResponse class was created for, so that clients can look at the status of the response from the server... If I set the response to null, then should I remove the DummyResponse class as well?